corehq/apps/export/views/list.py

```diff
@@ -56,52 +56,52 @@
 from corehq.apps.export.views.utils import ExportsPermissionsManager, user_can_view_deid_exports
 
 
-class BaseExportListView(HQJSONResponseMixin, BaseProjectDataView):
-    template_name = 'export/export_list.html'
-    allow_bulk_export = True
+class ExportListHelper(object):
+    '''
+    Encapsulates behavior that varies based on model (form, case, or sms)
+    and is needed by the function-based views in this module.
+    '''
+    form_or_case = None         # None if this handles both forms and cases
     is_deid = False
 
-    lead_text = ugettext_lazy('''
-        Exports are a way to download data in a variety of formats (CSV, Excel, etc.)
-        for use in third-party data analysis tools.
-    ''')
-
-    @use_select2
-    @method_decorator(login_and_domain_required)
-    def dispatch(self, request, *args, **kwargs):
-        self.permissions = ExportsPermissionsManager(self.form_or_case, request.domain, request.couch_user)
-        self.permissions.access_list_exports_or_404(is_deid=self.is_deid)
-
-        return super(BaseExportListView, self).dispatch(request, *args, **kwargs)
+    @classmethod
+    def get(self, request, domain, form_or_case=None, is_daily_saved_export=False, is_feed=False, is_deid=False):
+        if is_feed:
+            if is_deid:
+                return DeIdDashboardFeedListHelper(request, domain)
+            return DashboardFeedListHelper(request, domain)
+
+        if is_daily_saved_export:
+            if is_deid:
+                return DeIdDailySavedExportListHelper(request, domain)
+            return DailySavedExportListHelper(request, domain)
+
+        if form_or_case == 'form':
+            if is_deid:
+                return DeIdFormExportListHelper(request, domain)
+            return FormExportListHelper(request, domain)
+        elif form_or_case == 'case':
+            return CaseExportListHelper(request, domain)
```

Comment:
You could also set the appropriate helper class as a property of each view class and avoid needing to do this dispatch method (unless of course your plan is to eventually deduplicate those view classes).

Reply:
Not quite following - the only place I call this is in the `get_exports_page` functional view. How does having the helper be a property of the class-based list views help?

Reply:
Oh yeah, nevermind - I think my comment made more sense when the CBVs were using this method, but since there's only one functional view, this approach works better.